File: qqbot-client/src/main/java/com/mieai/qqbot/client/QqKeyboardMessageRequest.java
package com.mieai.qqbot.client;

import java.util.List;
import java.util.Map;
import java.util.Objects;
import java.util.Optional;

public record QqKeyboardMessageRequest(QqMessageTargetType targetType, String targetId,
        String markdownContent,
        Optional<String> keyboardId, List<Map<String, Object>> rows,
        Optional<String> replyMessageId, Optional<String> replyEventId, int messageSequence) {
    public QqKeyboardMessageRequest {
        Objects.requireNonNull(targetType, "targetType must not be null");
        if (targetId == null || targetId.isBlank()) throw new IllegalArgumentException("targetId is invalid");
        if (markdownContent == null || markdownContent.isBlank() || markdownContent.length() > 4000) {
            throw new IllegalArgumentException("markdownContent is invalid");
        }
        Objects.requireNonNull(keyboardId, "keyboardId must not be null");
        rows = rows == null ? List.of() : rows.stream().map(Map::copyOf).toList();
        if (keyboardId.isEmpty() && rows.isEmpty()) throw new IllegalArgumentException("keyboardId or rows is required");
        Objects.requireNonNull(replyMessageId, "replyMessageId must not be null");
        Objects.requireNonNull(replyEventId, "replyEventId must not be null");
        if (messageSequence < 1) throw new IllegalArgumentException("messageSequence must be positive");
    }
}

File: qqbot-client/src/main/kotlin/com/mieai/qqbot/client/QqKeyboardMessageRequest.kt
package com.mieai.qqbot.client

class QqKeyboardMessageRequest(
    val targetType: QqMessageTargetType,
    val targetId: String,
    val markdownContent: String,
    val keyboardId: String? = null,
    rows: List<Map<String, Any>> = emptyList(),
    val replyMessageId: String?,
    val replyEventId: String?,
    val messageSequence: Int,
) {
    val rows: List<Map<String, Any>> = rows.map { it.toMap() }

    init {
        require(targetId.isNotBlank()) { "targetId is invalid" }
        require(markdownContent.isNotBlank() && markdownContent.length <= 4000) {
            "markdownContent is invalid"
        }
        require(keyboardId != null || rows.isNotEmpty()) { "keyboardId or rows is required" }
        require(messageSequence >= 1) { "messageSequence must be positive" }
    }
}
